src/main/mdo/assembly-component.mdo

```diff
@@ -283,6 +283,20 @@
           <defaultValue>false</defaultValue>
           <type>boolean</type>
         </field>
+        <field>
+          <name>fileMappers</name>
+          <version>3.2.0+</version>
+          <association>
+            <type>FileMapper</type>
```

Comment:
You can't use the FileMapper from Plexus IO directly. You need to declare all types (except "simple" types such as String) in the MDO. So you have to declare this type in MDO as well and then use it to configure the actual FileMappers.